scarb/src/compiler/compilation_unit.rs

```diff
@@ -48,6 +48,8 @@ pub struct CompilationUnitComponent {
     pub package: Package,
     /// Information about the specific target to build, out of the possible targets in `package`.
     pub target: Target,
+    /// Items for the Cairo's `#[cfg(...)]` attribute to be enabled in this component.
+    pub cfg_set: CfgSet,
```

Comment:
By component-level cfg you mean cfg_set in crate configuration?
If that's so can you elaborate on setting in db because I don't quite get what you are saying.

Reply:
We have two level of cfg - one on `CompilationUnit` and then second one on `CompilationUnitComponent`. My understanding is, that `CompilationUnit` one should be a default, then if we want to change `cfg_set` for specific crate, then `CompilationUnitComponent` can override the default (lmk if it is not true). This would be in 1:1 relation to `RootDatabaseBuilder.with_cfg` and `CrateSettings.cfg_set` from Cairo compiler (once again - correct me if I am wrong).

The cfg_set from `CompilationUnit` is set to the Cairo compiler db (as the global one) with `with_cfg` here: https://github.com/software-mansion/scarb/blob/46d5d5ceac98ebb0a947d7e85024ede3aa5b4090/scarb/src/compiler/db.rs#L25C7-L25C15

I imagine, that with this PR we should also set `cfg_set` from `CompilationUnitComponent` in `CrateSettings` where applicable. However, since we already have the global `cfg_set` in db, I suggest that we can skip setting the `CompilationUnitComponent` one, if it is identical to the global one (which could be marked with `Option::None` in the `CompilationUnitComponent` struct).

Reply:
Yes, you are right in terms of how CompilationUnit/Component are related to CrateSettings and stuff. 

Initially only TestCompiler was supposed to use CrateSetting::cfg_set to override the behaviour of MacroPlugin but now that logic was moved to compilation units generation and I guess it makes sense to override those settings for each target. Do you know where is the best place in the code to do that?

When it comes to cfg_set being optional on components I'll mark them as such if that makes more sense.

Reply:
> now that logic was moved to compilation units generation and I guess it makes sense to override those settings for each target

👍 
Yep, I think this can be more general now. 

> Do you know where is the best place in the code to do that
I guess it goes there, right? 

https://github.com/software-mansion/scarb/blob/46d5d5ceac98ebb0a947d7e85024ede3aa5b4090/scarb/src/compiler/db.rs#L101